### DISCUSSION TITLE
SeaChest_Erase

### DISCUSSION THREAD

**Discussion by ghost** [Seagate/openSeaChest#105] [Q&A] [answered]
Does --performQuickestErase wipe HPA and DCO or I have to use --ataSecureErase enhanced mode.

**Comment by vonericsen** [comment, marked as answer]
Hi @Tokvin,

Thank you for this question!
If you look at the output of `--showEraseSupport` it will list the erases that openSeaChest_Erase or SeaChest_Erase will use to determine which erase to run.
The `--performQuickestErase` will use whatever is the first entry in that list that is output to do the erase.

If you wanted to help ensure and HPA/AMAC protected areas are wiped no matter what erase is used, you can use this command to try removing that protected area first: `openSeaChest_Configure -d <handle> --restoreMaxLBA`. After this is done anything in HPA or AMAC will then be accessible to wipe by any of the erases supported, not just Sanitize and enhanced ATA secure erase.
We do not currently have any commands to work with DCO, but we can look into adding those as well. 

As discussed in #52 both Sanitize and ATA enhanced secure erase will wipe out these areas without having to run any commands first. If your drive supports Sanitize, I recommend it over enhanced ATA security erase because it is not interruptible. 

I have had my own very different experiences with enhanced ATA security erase being interrupted and have had lots of reports about it, but I cannot find any other ways of preventing resets from happening. In the most recent report I had from inside Seagate , they started the erase on a drive connected to the motherboard, then they were trying to do something else to another drive connected to the same motherboard. As soon as they issued a command to the drive that was not erasing, the drive running the enhanced secure erase received a reset and stopped the erase. The only explanation I have for this behavior is either the hardware or the driver for this board cannot handle 2 drives in a "busy" state on the bus at the same time...so it resets the busy drive before issuing a command.
So, if possible, I would recommend only erasing one drive at a time and possibly doing it from a Live USB or something like that. This could prevent this problem since the OS drive will likely issue some commands to the OS drive while the drive being erased is still processing.

This is not an issue with the sanitize feature. It was designed to be started and then let the host poll from progress indication and to be completely uninterruptable. You can even remove the power and once the drive is powered back on it will resume the erase where it last left off. I have had a lot less trouble getting sanitize erases to complete than ATA security erase on HDDs just due to how long an ATA security erase takes on current HDDs.
